daml-assistant/daml-helper/src/DamlHelper/Run.hs

```diff
@@ -122,7 +121,7 @@ publishedExtensionName = "DigitalAssetHoldingsLLC.daml"
 -- | Name of VS Code extension bundled with the SDK. Legacy, but also
 -- useful in a pinch, in case published extension goes bad.
 bundledExtensionName :: String
-bundledExtensionName = "da-vscode-daml-extension"
+bundledExtensionName = "daml-bundled"
```

Comment:
This doesn't make sense to me, in the context of how the program uses `bundledExtensionName`. It was used in `getInstalledExtensions` (which I have no idea how to actually implement any more, since there isn't any distinction between the vsix package we bundle and the extension coming from the marketplace). I'm in favor of just scrapping that code and rewriting `runDamlStudio`  from scratch to have only the `--replace=never`, `--replace=always`, and `--replace=published` modes. 

In theory all that has to be done in `daml studio` is:

For `--replace=published`:

    1. install the extension based on the name.
    2. if this fails, install the bundled vsix
    3. if that fails, error.
    4. open code at the current project (or current dir if outside project)

For `--replace=always`:

    1. call `code` to uninstall the existing extension. Ignore exit code -- you don't care if the uninstall failed.
    2. call `code` to install the bundled vsix
    3. if that fails, error
    4. open code at the current project (or current dir if outside project)

For `--replace=never`:
 
    1. open code at the current project (or current dir if outside project)



----

I the following 8 step manual test process to make sure everything works:

1. quit vscode, run `daml studio --replace=always`, go to a daml file and make sure there is syntax highlighting
2. quit vscode, run `daml studio --replace=always`, go to a daml file and make sure there is syntax highlighting (again)
3. quit vscode, run `daml studio`, go to a daml file and make sure there is syntax highlighting
4. quit vscode, run `daml studio`, go to a daml file and make sure there is syntax highlighting (again)
5. turn off your wifi / internet connection
6. quit vscode, run `daml studio`, go to a daml file and make sure there is syntax highlighting
7. quit vscode, run `daml studio`, go to a daml file and make sure there is syntax highlighting (again)
8. turn back on your wifi / internet connection

Reply:
The bundled extension is slightly different: We give it a different name to avoid autoupgrades and to be able to figure out where we got the extension from. That said, the name should be `DigitalAssetHoldingsLLC.daml-bundled`. I think rewriting it from scratch is a bit overstating what needs to be done.

The logic for installing the published extension should stay the same. The logic for removing the bundled extension in the old format should also mostly stay the same but we need a new case to also remove the bundled extension in the new format.

Somewhat orthogonal to this PR but we should also add a `--replace=bundled` alias that behaves identically to `--replace=always`.